contracts/core/modules/IssuanceOrderModule.sol

```diff
@@ -405,48 +327,194 @@ contract IssuanceOrderModule is
+     * @return makerTokenUsed          Amount of maker token used to execute orders
      */
-    function settleOrder(
-        OrderLibrary.IssuanceOrder _order,
+    function executeExchangeOrders(
+        bytes _orderData,
+        address _makerAddress,
+        address _makerTokenAddress,
         uint256 _fillQuantity,
-        bytes _orderData
+        uint256 _attemptedFillQuantity
     )
         private
+        returns (uint256)
     {
-        // Declare IVault interface as variable
-        IVault vaultInstance = IVault(vault);
+        uint256 scannedBytes;
+        uint256 makerTokenUsed;
+        while (scannedBytes < _orderData.length) {
 
-        // Check to make sure open order amount equals _fillQuantity
-        uint256 closedOrderAmount = orderFills[_order.orderHash].add(orderCancels[_order.orderHash]);
+            // Parse next exchange header based on scannedBytes
+            ExchangeHeaderLibrary.ExchangeHeader memory header = ExchangeHeaderLibrary.parseExchangeHeader(
+                _orderData,
+                scannedBytes
+            );
 
-        // Open order amount is greater than or equal to closed order amount
-        require(
-            _order.quantity.sub(closedOrderAmount) >= _fillQuantity,
-            "IssuanceOrderModule.settleOrder: Fill amount exceeds order available quantity"
+            // Get exchange address from state mapping based on header exchange info
+            address exchangeWrapper = ICore(core).exchanges(header.exchange);
+
+            // Verify exchange address is registered
+            require(
+                exchangeWrapper != address(0),
+                "IssuanceOrderModule.executeExchangeOrders: Invalid or disabled Exchange address"
+            );
+
+            // Read the order body based on order data length info in header plus the length of the header (128)
+            uint256 exchangeDataLength = header.orderDataBytesLength.add(128);
+            bytes memory bodyData = LibBytes.slice(
+                _orderData,
+                scannedBytes.add(128),
+                scannedBytes.add(exchangeDataLength)
+            );
+
+            // Calculate amount of makerToken actually needed
+            uint256 neededMakerTokenAmount = OrderLibrary.getPartialAmount(
+                header.makerTokenAmount,
+                _fillQuantity,
+                _attemptedFillQuantity
+            );
+
+            // Transfer maker token to Exchange Wrapper to execute exchange orders
+            // Using maker token from signed issuance order to prevent malicious encoding of another maker token
+            ITransferProxy(transferProxy).transfer(
+                _makerTokenAddress,
+                neededMakerTokenAmount,
+                _makerAddress,
+                exchangeWrapper
+            );
+
+            // Call Exchange
+            callExchange(
+                [_makerAddress, msg.sender, _makerTokenAddress],
```

Comment:
Would the `taker` ever not be `msg.sender`?

Reply:
Should we even pass msg.sender into callExchange vs passing `msg.sender` in call exchange to the exchange wrapper?